wait_test.go

```diff
@@ -161,4 +163,74 @@ func Test_Wait(t *testing.T) {
 		}
 		require.EqualError(t, err, fmt.Sprintf("timed out connecting to %q", wantAddr))
 	})
+
+	t.Run("success unix", func(t *testing.T) {
+		t.Parallel()
+
+		var gotRequest bool
+		t.Cleanup(func() {
+			assert.True(t, gotRequest, "wanted unix server to receive request")
+		})
+
+		socket := newTestUnixServer(t, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			gotRequest = true
+			assert.Equal(t, "/foo/bar", r.URL.Path)
+			assert.Empty(t, r.URL.Host)
+
+			w.WriteHeader(http.StatusOK)
+		}))
+
+		ctx := context.Background()
+		err := grace.Wait(
+			ctx,
+			50*time.Millisecond,
+			grace.WithWaitForUnix(socket),
+			grace.WithWaitForUnixHTTP(socket, "/foo/bar"),
```

Comment:
Do we not want separate tests for unix vs unix http? 

In particular I don't see any test cases where `WaitForUnix` fails, but in general I think dedicated tests for each function makes sense.

Reply:
I think that you are correct. Added.